op-challenger/fault/monitor.go

```diff
@@ -78,7 +79,15 @@ func (m *gameMonitor) progressGames(ctx context.Context) error {
 			m.logger.Error("Error while progressing game", "game", game.Proxy, "err", err)
 			continue
 		}
-		player.ProgressGame(ctx)
+		done := player.ProgressGame(ctx)
+		if done {
+			// Remove resources on disk as soon as the game is complete to save disk space.
+			// We keep the player in memory to avoid recreating it on every update but will no longer
+			// need the resources on disk because there are no further actions required on the game.
+			if err := player.Cleanup(); err != nil {
+				m.logger.Error("Unable to cleanup player data", "err", err)
```

Comment:
Interesting, although this shouldn't fail, if there's a bug, it's going to keep printing this log every time the game is attempted to progress (concretely, every time the `progressGames` method on the monitor is called) because the game player will always return `true` from the `done := player.ProgressGame(ctx)` call.

Reply:
Yeah that will be messy but I think spamming the logs is probably better than failing to clean up once and giving up and just leaking the disk space.

Reply:
Agree - it's explicitly an error as well, not like we're cluttering logs with dismiss-able warnings